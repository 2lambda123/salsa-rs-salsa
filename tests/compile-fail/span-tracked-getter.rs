#[salsa::tracked]
pub struct MyTracked<'db> {
    field: u32,
}

#[salsa::tracked]
fn my_fn(db: &dyn salsa::Database) -> salsa::Result<()> {
    let x = MyTracked::new(db, 22)?;
    x.field(22)?;
    Ok(())
}

fn main() {
    let mut db = salsa::DatabaseImpl::new();
    my_fn(&db).unwrap();
}
