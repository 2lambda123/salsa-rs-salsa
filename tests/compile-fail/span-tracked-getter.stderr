error[E0308]: mismatched types
  --> tests/compile-fail/span-tracked-getter.rs:24:13
   |
24 |     x.field(22);
   |       ----- ^^ expected `&dyn Db`, found integer
   |       |
   |       arguments to this method are incorrect
   |
   = note: expected reference `&(dyn Db + 'static)`
                   found type `{integer}`
note: method defined here
  --> tests/compile-fail/span-tracked-getter.rs:18:5
   |
16 | #[salsa::tracked]
   | -----------------
17 | pub struct MyTracked<'db> {
18 |     field: u32,
   |     ^^^^^
